src/box/sql/build.c

```diff
@@ -2051,20 +2051,33 @@ sql_drop_table_constraint(struct Parse *parser, const struct Token *table_name,
 		sqlVdbeCountChanges(v);
 		sqlVdbeAddOp4(v, OP_DropTupleForeignKey, space->def->id, 0, 0,
 			      name_str, P4_DYNAMIC);
-		return;
 	}
 
 	const struct tuple_constraint_def *ck =
 		sql_tuple_ck_by_token(space, name);
 	if (ck != NULL) {
+		if (fk != NULL) {
+			diag_set(ClientError, ER_SQL_EXECUTE,
+				 tt_sprintf("ambiguous constraint name: '%s'",
+					    name_str));
+			parser->is_aborted = true;
+			return;
```

Comment:
Leak of `name_str` occurs.

Reply:
No, it was already assigned to an opcode.